### ISSUE TITLE
False positive with cljs.core/set!

### ISSUE THREAD

**Issue by kommen** [borkdude/clj-kondo#530]
**version**

clj-kondo v2019.10.11-alpha

**platform**

macOS 10.15

**problem**

As @mfikes [points out](https://twitter.com/mfikes/status/1181708123511373827) [`cljs.core/set!`](https://cljs.github.io/api/cljs.core/setBANG) has a variant using an instance field name:

```clj
(set! (js-obj "a" 1) -a 2)
```

clj-kondo reports an unresolved symbol error, which it probably shouldn't.

**repro**

```
$ echo '(set! (js-obj "a" 1) -a 2)' | clj-kondo --lang cljs --lint -
```

**expected behavior**

No error reported.

**actual behavior**

Error reported:
```
<stdin>:1:22: error: unresolved symbol -a
```

**Comment by borkdude** [comment]
Thanks for reporting. Out of curiosity, is this used anywhere and are you using this variant?

**Comment by kommen** [comment]
I've used it once, but only since Mike mentioned it. I'm not aware of any "real world" usage.
